### PR TITLE
Add scANVI training script

### PR DESCRIPTION
**Pull request by sjspielman**
#### Please link to the GitHub issue that this pull request addresses.

Towards #1214, #1246

#### What is the goal of this pull request?

This PR adds a new script to train a scANVI model on NBAtlas and export it for label transfer coming up in the next PR. I also updated an existing script to allow for either a merged or single-library SCE object to be passed in to scANVI.

#### Briefly describe the general approach you took to achieve this goal.

1. Added `scripts/03a_train-scanvi-model.py`
  - This script trains a SCVI model and then trains a scANVI model from that, which is recommended for scANVI. Parameters for scArches are also included, as are two categorical covariates representing the suspension type and the 10x tech version. 
  - The script exports the trained SCVI model, the trained scANVI model, and a standalone TSV of the scANVI latent representation for the NBAtlas object which may be useful for downstream visualization without needing to dig into the model object innards
2. Modified `scripts/03b_prepare-scanvi-query.R`
  - This script used to be `03a`, but I swapped it to be `03b` since if/when we move towards running scANVI with single libraries, this will be run second in order. The a/b order doesn't matter much now.
  - I updated this script to include a `--merged` flag to indicate whether the object being passed in is merged, since this affects how to update the colData.
3. Adds docs to `scripts/README`
4. Added these steps to `run-analysis.sh` and activated the conda env in the GHA

#### What are the software and computational requirements needed to be able to run the code in this PR?

Runs on a laptop!

#### Are there particularly areas you'd like reviewers to have a close look at?

N/A

#### Is there anything that you want to discuss further?

N/A

#### Analysis module and review

- [x] This analysis module [uses the analysis template and has the expected directory structure](https://openscpca.readthedocs.io/en/latest/contributing-to-analyses/analysis-modules/).
- [x] The analysis module `README.md` has been updated to reflect code changes in this pull request.
- [x] The analytical code is documented and contains comments.
- [ ] Any results and/or plots this code produces have been added to your S3 bucket for review.

#### Reproducibility checklist

- [x] Code in this pull request has been added to the GitHub Action workflow that runs this module.
- [ ] The dependencies required to run the code in this pull request have been added to the analysis module `Dockerfile`.
- [ ] If applicable, the dependencies required to run the code in this pull request have been added to the analysis module conda `environment.yml` file.
- [ ] If applicable, R package dependencies required to run the code in this pull request have been added to the analysis module `renv.lock` file.

### CODE REVIEW

**Review by jashapiro — COMMENTED** [review]
Looks good! I made some "pythonic" suggestions and a question about defaults, but nothing too major.

File: analyses/cell-type-neuroblastoma-04/scripts/03a_train-scanvi-model.py [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+import os
+import sys
+from pathlib import Path
+import argparse
+import anndata
+import scvi
+import torch
+import pandas as pd
+from scipy.sparse import csr_matrix
```
[the remark this answers is not in the document]
Comment (jashapiro) [reply analyses/cell-type-neuroblastoma-04/scripts/03a_train-scanvi-model.py:14]: Apparently it doesn't format by default: https://docs.astral.sh/ruff/formatter/#sorting-imports

But linting is a good idea too!

File: analyses/cell-type-neuroblastoma-04/scripts/03a_train-scanvi-model.py [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+    )
+    arg = parser.parse_args()
+
+    ################################################
+    ########### Input argument checks ##############
+    ################################################
+    arg_error = False
+
+    # Check that input files exist
+    if not os.path.isfile(arg.reference_file):
```
[the remark this answers is not in the document]
Comment (sjspielman) [reply analyses/cell-type-neuroblastoma-04/scripts/03a_train-scanvi-model.py]: Actually they aren't dirs yet - `scvi` creates them when writing out. It creates a directory with that name and stores files auto-named `model.pt` and `adata.h5ad` (latter if specified, and I do specify) in it.

**Comment by sjspielman** [comment]
I think I've got it all, ready for another look!
